lib/agent0/agent0/hyperdrive/exec/trade_loop.py

```diff
@@ -76,32 +76,22 @@ def trade_if_new_block(
                     float(trade_result.trade_object.market_action.trade_amount),
                 )
             elif trade_result.status == TradeStatus.FAIL:
-                # To detect slippage, we look for the `OutputLimit` exception thrown from the smart contract.
-                # Since this exception is used elsewhere (e.g., in redeem withdraw shares), we also explicitly check
-                # that the trade here is open/close long/short.
-                # TODO this error is not guaranteed to be exclusive for slippage in the future.
-                is_slippage = (
-                    isinstance(trade_result.exception, ContractCustomError)
-                    and ("OutputLimit raised" in trade_result.exception.args[1])
-                    and (
-                        trade_result.trade_object.market_action.action_type
-                        in (
-                            HyperdriveActionType.OPEN_LONG,
-                            HyperdriveActionType.CLOSE_LONG,
-                            HyperdriveActionType.OPEN_SHORT,
-                            HyperdriveActionType.CLOSE_SHORT,
-                        )
-                    )
-                )
+                # Here, we check for common errors and allow for custom handling of various errors
+
+                # Detecting invalid balance issues here
```

Comment:
add another comment explaining that these functions modify the `trade_result` to include whether or not there was an invalid_balance or slippage